### ISSUE TITLE
Can somebody explain me how CSRF are validated?

### ISSUE THREAD

**Issue by advename** [jazzband/dj-rest-auth#40]
I can't find anywhere in the documentation how CSRF are validated, but by inspecting the Headers, 
`csrftoken` and `sessionid` are still send.

Usually, you would place `{% csrf_token %}` in the form.

**Comment by iMerica** [comment]
Hi @advename,

CSRF is not in scope in this package because: 
1) This package is for authentication in Django **REST** Framework, not Django itself (classic server rendered web apps) - it's an entirely different paradigm with different threat models. 
2) This is an API authentication package, not an everything-you-possibly-need-to-secure-your-API python package. People who use this package will often use separate packages for things like handling cors headers, input sanitization, 2FA and other security concerns. 

I do understand the concern though, so let me explain a few classic CSRF examples and how they play out in a SPA <~> API scenario.

**Site**
example.com

**Server**
A DRF API that uses this package with default functionality (Token-based auth)

**Client**
A React SPA in a browser. The user stores the auth token in local storage.
____
**Scenario A**
- A user is authenticated into the site (the server/client described above). 
- The user receives a malicious email with a link to a fraudulent site of the site above.
- The fraudulent site has an action button to the genuine site. Imagine this button is a GET request to the genuine site that attempts to do something malicious.
- The user clicks the button.

**Results**
By clicking the link the user has performed a GET request to the site. This action alone doesn't do anything but load a dumb HTML document that includes a JS bundle that will render a page. If something destructive is happening from a GET request, [then all bets are off](https://softwareengineering.stackexchange.com/questions/188860/why-shouldnt-a-get-request-change-data-on-the-server). 
______
**Scenario B**
Same as Scenario A, but instead of the link directing to example.com where a static web server is listening, it points to api.example.com, where DRF is listening.

**Results**
The request is blocked by CORS / same-origin policy.
______
**Scenario C**
Same as scenario A, but using JWT Tokens stored in a browser using an HTTP-only cookie. The malicious site is directing to request to api.example.com, where DRF is listening.

**Results**
No harm done. You just did a GET (read-only) request on a site that you have already successfully authenticated to. 
______
**Scenario D**
Same as scenario A, but using JWT Tokens stored in a browser using an HTTP-only cookie. The malicious site has a button that does a POST request to api.example.com when clicked. 

```html
<html lang="">
    <head>
        <script type="application/javascript">
            function fn() {
              fetch('https://api.example.com/account/transfer/', {
                method: 'POST',
                body: JSON.stringify({
                  // some destructive payload
                }),
                mode: "no-cors"
              })
            }
        </script>
        <title>CSRF Attack</title>
    </head>
    <body>
        <button onclick="fn()">Click Me!</button>
    </body>
</html>
```

**Results**

Http 403 because no authentication details were provided.
